File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-0.java
class Solution {
    public int findMin(int[] nums) {
        int l =0;
        int r = nums.length - 1;

        if (nums[l] < nums[r]) {
            return nums[l];
        }

        while (l < r) {
            if (nums[l+1] > nums[l]) {
                l++;
            } else {
                return nums[l+1];
            }

            if (nums[r-1] < nums[r]) {
                r--;
            } else {
                return nums[r];
            }
        }

        return Math.min(nums[r],nums[l]);
    }
}

File: Data Structures & Algorithms/maximum-subarray/submission-0.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        maxSum = sys.maxsize * -1
        curSum = 0

        for num in nums:
            if curSum < 0:
                curSum = 0

            curSum += num
            maxSum = max(maxSum, curSum)

        return maxSum
